spec/services/broadcasts/welcome_notification/generator_spec.rb

```diff
@@ -10,10 +18,51 @@
     end
 
     context "when sending a welcome_thread notification" do
-      xit "generates the appropriate broadcast to be sent to a user"
-      xit "it sends a welcome notification for that broadcast"
-      xit "it does not send duplicate welcome notification for that broadcast"
-      xit "does not send a notification to a user who has commented in a welcome thread"
+      let(:welcome_thread_article) { create(:article, title: "Welcome Thread") }
+      let(:welcome_thread_comment) { create(:comment, commentable: welcome_thread_article, user: user) }
+      let(:article) { create(:article) }
+      let(:comment) { create(:comment, commentable: article, user: receiving_user) }
+
+      it "generates the correct broadcast type and sends the notification to the user", :aggregate_failures do
```

Comment:
Question for reviewers: Is there a convention for using or not using `:aggregate_failures` for multiple expectations in specs?

Reply:
There's no convention that I'm aware of. It's case by case and really depends on what you are testing for. Folks tend to use them [when testing/comparing for multiple values](https://makandracards.com/makandra/65467-rspec-how-to-aggregate-failures).

In this spec here though, I don't think you need it because you are doing a before-and-after test. You could rely on `expect {..}.to have change(receiving_user.notifications, :count).by(1)` to simplify it. Or perhaps testing for `count` here is counterproductive and the only thing we need here is to make sure `welcome_broadcast` is properly broadcasted.